### ISSUE TITLE
A direction from an operation to another doesn't match

### ISSUE THREAD

**Issue by chatterboy** [tensorflow/tensorboard#1538]
TensorBoard version: 1.7.0
TensorFlow version: 1.7.0
OS Platform and version: Windows 10 Education 64bits
Python version: 3.5.0

I was running a simple example about MNIST with MLP on tensorflow and saving the logs using "tf.summary.FileWriter". After that I run the tensorboard to visualize something in the logs. I was looking at the graphs. But, in the variable operations the direction is wrong compared to the description. Here is an example:

![image](https://user-images.githubusercontent.com/9608328/47340384-08483f80-d6d9-11e8-973f-5f39f17430dc.png)

As you can see, in the description the "Variable_1/(Variable_1)" operation have no inputs and three outputs. But, In the graphs that operation have one input and two outputs.

**Comment by stephanwlee** [comment]
Hi there! Where can I find the example MNIST with MLP? I'd like to run it myself to debug the issue. Thanks!

**Comment by chatterboy** [comment]
@stephanwlee 

I modified some parts of the code as i was forgetting to submit the issue. Anyway the issue is in present still. Thank you.

`
import os
import tensorflow as tf
os.environ['TF_CPP_MIN_LOG_LEVEL'] = '2'

from tensorflow.examples.tutorials.mnist import input_data
mnist = input_data.read_data_sets('data', one_hot=True)

height = 28
width = 28
batch_size = 100
num_epochs = 5000
num_classes = 10
display_cycles = 200
learning_rate = 0.001

x = tf.placeholder(tf.float32, shape=[None, height * width])
y = tf.placeholder(tf.float32, shape=[None, num_classes])

W = tf.Variable(tf.zeros([height * width, num_classes]))
b = tf.Variable(tf.zeros([num_classes]))

def NN(x, W, b):
    return tf.matmul(x, W) + b

def softmax_cross_entropy_loss(logits, labels):
    p = tf.nn.softmax(logits, axis=-1)
    ce = labels * tf.log(p)
    return -tf.reduce_mean(tf.reduce_sum(ce, axis=1))

def get_accuracy(logits, labels):
    p = tf.nn.softmax(logits, axis=-1)
    return tf.reduce_mean(tf.cast(tf.equal(tf.argmax(p, axis=1), tf.argmax(labels, axis=1)), tf.float32))

logits = NN(x, W, b)

accuracy = get_accuracy(logits=logits, labels=y)

loss = softmax_cross_entropy_loss(logits=logits, labels=y)

train = tf.train.AdamOptimizer(learning_rate=learning_rate).minimize(loss)

with tf.Session() as sess:
    with tf.summary.FileWriter('./graphs', sess.graph) as writer:
        sess.run(tf.global_variables_initializer())

        # Train phase
        for epoch in range(1, num_epochs + 1):
            batch_x, batch_y = mnist.train.next_batch(batch_size)

            loss_, accuracy_, _ = sess.run([loss, accuracy, train], feed_dict={x:batch_x, y:batch_y})

            if epoch % display_cycles == 0:
                print("epoch: {} - loss: {} - accuracy: {}".format(epoch, loss_, accuracy_))

        # Test phase
        loss_, accuracy_ = sess.run([loss, accuracy], feed_dict={x:mnist.test.images, y:mnist.test.labels})

        print("Test phase - loss: {} - accuracy: {}".format(loss_, accuracy_))
`

**Comment by stephanwlee** [comment]
Thanks @chatterboy for the example. It was really helpful.

For posterity, I will leave some of my findings here as I am not the original author:
It looks like the edges appear "incorrect" for all Assign ops. From https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/ops/state_ops.cc#L80-L82, it looks like Assign ops are defined to take Ref(A) and output to Ref(A) and bidirectional edge, creating a cycle. Instead of drawing the cycle, it looks like TB only has one edge based on this definition here:
https://github.com/tensorflow/tensorboard/blob/master/tensorboard/plugins/graph/tf_graph_loader/tf-graph-loader.html#L145-L157
The drawing code is here https://github.com/tensorflow/tensorboard/blob/master/tensorboard/plugins/graph/tf_graph_common/edge.ts#L316

Now, it looks reasonable that, for Node/Ops like Assign Ops, the edge goes toward the reference node.  I think the correct fix is to change how information box depict the input/output instead. 

@nfelt do you have any context on this code?

**Comment by nfelt** [comment]
I don't have any special context here, but I agree that drawing the edge from the Assign op to the variable seems a little more intuitive to me than vice versa, so updating Inputs/Outputs SGTM.  We could also ask the original graph visualization authors if they have input.
